PeerDB-io/peerdb
# feat: Implement OpenTelemetry tracing

- Add basic telemetry and examples
- Integrate with temporal's opentelemetry interceptor


To run/test:
1. Run jaeger:
   ```shell
   docker run --rm --name jaeger -p 16686:16686 -p 4317:4317 -e JAEGER_REPORTER_LOG_SPANS=true -e JAEGER_SAMPLER_TYPE=const -e JAEGER_SAMPLER_PARAM=1 -e COLLECTOR_OTLP_ENABLED=true jaegertracing/all-in-one:1.57
   ```
   Open jaeger ui at http://localhost:16686/
2. Run flow worker with the following env:
   ```shell
   ENABLE_OTEL_TRACING=true
   OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=grpc
   OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=http://localhost:4317
   ```
Start any activity etc and we should start seeing traces like below:
<img width="1672" alt="image" src="https://github.com/PeerDB-io/peerdb/assets/39487888/ab5fc96c-7bc4-409a-9ec4-41c42069f5f5">
